smolyay/surrogate.py

```diff
@@ -192,13 +192,20 @@ def gradient(self, x):
         """
         if self._coefficients is None:
             raise ValueError('Function needs training!')
+        input_shape = numpy.shape(x)
         x = numpy.array(x, copy=False, ndmin=2)
         if self.dimension == 1:
             if x.ndim == 2 and x.shape[0] == 1 and x.shape[1]  > 1:
                 # the cast to 2d puts these in wrong order, so transpose
                 x = x.T
-            else:
+            elif x.shape[-1] > 1:
```

Comment:
Yes. In the case of an input shaped like (4,3,2,1), adding an extra dimension is unnecessary. Adding a dimension is explicitly for the case where the x.shape[-1] != self.dimension check would fail when self.dimension is 1, which should never happen because a 1D function should always pass that check.